src/macaron/slsa_analyzer/provenance/intoto/v01/__init__.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module handles in-toto version 0.1 attestations."""
+
+from __future__ import annotations
+
+from typing import TypedDict, TypeGuard
+
+from macaron.slsa_analyzer.provenance.intoto.errors import ValidateInTotoPayloadError
+from macaron.util import JsonType
+
+
+class InTotoStatement(TypedDict):
+    """An in-toto version 0.1 statement.
+
+    This is the type of the payload in an in-toto version 0.1 attestation.
+    Specification: https://github.com/in-toto/attestation/tree/main/spec/v0.1.0#statement.
+    """
+
+    _type: str
+    subject: list[InTotoSubject]
+    predicateType: str  # noqa: N815
+    predicate: dict[str, JsonType] | None
+
+
+class InTotoSubject(TypedDict):
+    """An in-toto subject.
+
+    Specification: https://github.com/in-toto/attestation/tree/main/spec/v0.1.0#statement.
+    """
+
+    name: str
+    digest: dict[str, str]
+
+
+def validate_intoto_statement(payload: dict[str, JsonType]) -> TypeGuard[InTotoStatement]:
+    """Validate the statement of an in-toto attestation.
+
+    Specification: https://github.com/in-toto/attestation/tree/main/spec/v0.1.0#statement.
+
+    Parameters
+    ----------
+    payload : dict[str, JsonType]
+        The JSON statement after being base64-decoded.
+
+    Returns
+    -------
+    TypeGuard[InTotoStatement]
+        ``True`` if the subject element is valid, in which case its type is narrowed to an
+        ``InTotoStatement``; ``False`` otherwise.
```

Comment:
Shouldn't this be the following?
```suggestion
        ``True`` if the attestation statement is valid, in which case its type is narrowed to an
        ``InTotoStatement``; ``False`` otherwise.
```

Reply:
Thanks. This must be a copy-paste error. Fixed in 1a6d6ca.